app/utils/text-adjustments.ts

```diff
@@ -0,0 +1,46 @@
+import { getTextWidth } from "./get-text-width";
+
+const WORD_SPLITTING_THRESHOLD = 0.75;
+
+export const splitTextByWidth = (
+  text: string,
+  width: number,
+  options: { fontSize: number }
+): string => {
```

Comment:
@noahonyejese I wonder if you considered adding a `word-break` (`break-word`, `break-all`) CSS property to the axis titles? As they are regular `span`s, it's possible to get this behavior without having a custom wrapping function 🤔